DataChannel crash in master test

```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x798be4]



goroutine 2211 [running]:

github.com/pion/sctp.(*Association).OpenStream(0x0, 0x3500000001, 0x0, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/pion/sctp@v1.7.12/association.go:1300 +0x64
github.com/pion/datachannel.Dial(0x0, 0x1, 0xc0002abc68, 0x0, 0x0, 0x43978f)
	/home/runner/go/pkg/mod/github.com/pion/datachannel@v1.4.21/datachannel.go:73 +0x4f
github.com/pion/webrtc/v3.(*DataChannel).open(0xc00035a000, 0xc0001bc1e0, 0x1c, 0xc0000f6048)
	/home/runner/work/webrtc/webrtc/datachannel.go:164 +0x427
github.com/pion/webrtc/v3.(*PeerConnection).startSCTP(0xc00026e380)
	/home/runner/work/webrtc/webrtc/peerconnection.go:1323 +0x285
github.com/pion/webrtc/v3.(*PeerConnection).startRTP(0xc00026e380, 0x0, 0xc00000f1c0, 0xc000010528, 0x1, 0x1)
	/home/runner/work/webrtc/webrtc/peerconnection.go:2121 +0x9e5
github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription.func2()
	/home/runner/work/webrtc/webrtc/peerconnection.go:1149 +0x22c
github.com/pion/webrtc/v3.(*operations).start(0xc000200030)
	/home/runner/work/webrtc/webrtc/operations.go:84 +0xd5
created by github.com/pion/webrtc/v3.(*operations).Enqueue
	/home/runner/work/webrtc/webrtc/operations.go:35 +0x1ec
exit status 2
FAIL	github.com/pion/webrtc/v3	17.231s
FAIL
exit status 1
```

I managed to write a test to get the error, but this may not be the only way to get the error. To increase the chance of getting a race error, add time.Sleep in [datachannel.go:163](https://github.com/pion/webrtc/blob/38d35c6ff04b65e0a17b102c04902cf3f0e49a5a/datachannel.go#L163)
```go
func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
	...
	if d.id == nil {
		err := d.sctpTransport.generateAndSetDataChannelID(d.sctpTransport.dtlsTransport.role(), &d.id)
		if err != nil {
			return err
		}
	}
	time.Sleep(100 * time.Millisecond) // Here datachannel.go:163
	dc, err := datachannel.Dial(d.sctpTransport.association, *d.id, cfg)
	if err != nil {
		d.mu.Unlock()
		return err
	}
	...
}
```

Test
```go
func TestIssue1730(t *testing.T) {
	for i := 0; i < 1000; i++ {
		pcOffer, pcAnswer, err := newPair()
		assert.NoError(t, err)
		assert.NoError(t, signalPair(pcOffer, pcAnswer))
		wg := sync.WaitGroup{}
		wg.Add(2)
		go func() {
			pcOffer.CreateDataChannel("issue-1730", &DataChannelInit{})
			wg.Done()
		}()
		go func() {
			time.Sleep(50 * time.Millisecond)
			pcOffer.Close()
			wg.Done()
		}()
		wg.Wait()
	}
}
```

Result
```log
==================
WARNING: DATA RACE
Read at 0x00c00010f628 by goroutine 66:
  github.com/pion/webrtc/v3.(*DataChannel).open()
      D:/projects/github.com/pion/webrtc/datachannel.go:165 +0x2f5
  github.com/pion/webrtc/v3.(*PeerConnection).startSCTP()
      D:/projects/github.com/pion/webrtc/peerconnection.go:1326 +0x23a
  github.com/pion/webrtc/v3.(*PeerConnection).startRTP()
      D:/projects/github.com/pion/webrtc/peerconnection.go:2111 +0xcfd
  github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription.func2()
      D:/projects/github.com/pion/webrtc/peerconnection.go:1152 +0x1f6
  github.com/pion/webrtc/v3.(*operations).start()
      D:/projects/github.com/pion/webrtc/operations.go:84 +0x85

Previous write at 0x00c00010f628 by goroutine 68:
  github.com/pion/webrtc/v3.(*SCTPTransport).Stop()
      D:/projects/github.com/pion/webrtc/sctptransport.go:134 +0xfd
  github.com/pion/webrtc/v3.(*PeerConnection).Close()
      D:/projects/github.com/pion/webrtc/peerconnection.go:1881 +0x9d0
  github.com/pion/webrtc/v3.TestIssue1730.func2()
      D:/projects/github.com/pion/webrtc/peerconnection_test.go:667 +0x52

Goroutine 66 (running) created at:
  github.com/pion/webrtc/v3.(*operations).Enqueue()
      D:/projects/github.com/pion/webrtc/operations.go:35 +0x1a4
  github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription()
      D:/projects/github.com/pion/webrtc/peerconnection.go:1149 +0xc84
  github.com/pion/webrtc/v3.signalPair()
      D:/projects/github.com/pion/webrtc/peerconnection_test.go:61 +0x385
  github.com/pion/webrtc/v3.TestIssue1730()
      D:/projects/github.com/pion/webrtc/peerconnection_test.go:658 +0x9c
  testing.tRunner()
      C:/Users/dell/sdk/go1.15.10/src/testing/testing.go:1123 +0x202

Goroutine 68 (finished) created at:
  github.com/pion/webrtc/v3.TestIssue1730()
      D:/projects/github.com/pion/webrtc/peerconnection_test.go:665 +0x17b
  testing.tRunner()
      C:/Users/dell/sdk/go1.15.10/src/testing/testing.go:1123 +0x202
==================
--- FAIL: TestIssue1730 (2.16s)
    testing.go:1038: race detected during execution of test
FAIL
FAIL    github.com/pion/webrtc/v3       2.247s
FAIL

```

I have no idea how to fix this)

--- a/datachannel.go
+++ b/datachannel.go
@@ -104,19 +104,22 @@ func (api *API) newDataChannel(params *DataChannelParameters, log logging.Levele
 
 // open opens the datachannel over the sctp transport
 func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
+	if sctpTransport == nil {
+		return errSCTPNotEstablished
+	}
+	sctpTransport.lock.RLock()
+	association := sctpTransport.association
+	sctpTransport.lock.RUnlock()
+	if association == nil {
+		return errSCTPNotEstablished
+	}
+
 	d.mu.Lock()
-	if d.sctpTransport != nil {
-		// already open
+	if d.sctpTransport != nil { // already open
 		d.mu.Unlock()
 		return nil
 	}
 	d.sctpTransport = sctpTransport
-
-	if err := d.ensureSCTP(); err != nil {
-		d.mu.Unlock()
-		return err
-	}
-
 	var channelType datachannel.ChannelType
 	var reliabilityParameter uint32
 
@@ -160,8 +163,7 @@ func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
 			return err
 		}
 	}
-
-	dc, err := datachannel.Dial(d.sctpTransport.association, *d.id, cfg)
+	dc, err := datachannel.Dial(association, *d.id, cfg)
 	if err != nil {
 		d.mu.Unlock()
 		return err
@@ -173,19 +175,6 @@ func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
 	d.mu.Unlock()
 
 	d.handleOpen(dc)
-	return nil
-}
-
-func (d *DataChannel) ensureSCTP() error {
-	if d.sctpTransport == nil {
-		return errSCTPNotEstablished
-	}
-
-	d.sctpTransport.lock.RLock()
-	defer d.sctpTransport.lock.RUnlock()
-	if d.sctpTransport.association == nil {
-		return errSCTPNotEstablished
-	}
 	return nil
 }
 

--- a/sctptransport.go
+++ b/sctptransport.go
@@ -96,12 +96,13 @@ func (r *SCTPTransport) Start(remoteCaps SCTPCapabilities) error {
 	}
 	r.isStarted = true
 
-	if err := r.ensureDTLS(); err != nil {
-		return err
+	dtlsTransport := r.Transport()
+	if dtlsTransport == nil || dtlsTransport.conn == nil {
+		return errSCTPTransportDTLS
 	}
 
 	sctpAssociation, err := sctp.Client(sctp.Config{
-		NetConn:       r.Transport().conn,
+		NetConn:       dtlsTransport.conn,
 		LoggerFactory: r.api.settingEngine.LoggerFactory,
 	})
 	if err != nil {
@@ -133,15 +134,6 @@ func (r *SCTPTransport) Stop() error {
 
 	r.association = nil
 	r.state = SCTPTransportStateClosed
-
-	return nil
-}
-
-func (r *SCTPTransport) ensureDTLS() error {
-	dtlsTransport := r.Transport()
-	if dtlsTransport == nil || dtlsTransport.conn == nil {
-		return errSCTPTransportDTLS
-	}
 
 	return nil
 }

